File: test-api.py
import urllib.request
import json

data = json.dumps({"title": "Test Task", "description": "Test"}).encode('utf-8')
req = urllib.request.Request('http://localhost:3000/api/tasks/', data=data, headers={'Content-Type': 'application/json'})

try:
    with urllib.request.urlopen(req) as response:
        print(response.read().decode('utf-8'))
except urllib.error.HTTPError as e:
    print(f"HTTPError: {e.code} {e.reason}")
    print(e.read().decode('utf-8'))
except Exception as e:
    print(f"Error: {e}")

File: test-api.js
import axios from 'axios';

axios.post('http://127.0.0.1:8000/api/tasks/', { title: 'Test', description: 'Test' })
  .then(res => console.log('Success:', res.status, res.data))
  .catch(err => console.error('Error:', err.response ? err.response.status + ' ' + JSON.stringify(err.response.data) : err.message));
